pages/[network]/[chain]/pull-request.tsx

```diff
@@ -15,65 +15,138 @@ import NothingFound from "components/nothing-found";
 import PullRequestHero from "components/pull-request-hero";
 import ReadOnlyButtonWrapper from "components/read-only-button-wrapper";
 
-import {useAppState} from "contexts/app-state";
-import {BountyEffectsProvider} from "contexts/bounty-effects";
-import {changeCurrentBountyComments, changeCurrentBountyData} from "contexts/reducers/change-current-bounty";
-import {changeLoadState} from "contexts/reducers/change-load";
-import {changeSpinners} from "contexts/reducers/change-spinners";
-import {addToast} from "contexts/reducers/change-toaster";
+import { useAppState } from "contexts/app-state";
+import { BountyEffectsProvider } from "contexts/bounty-effects";
+import { changeCurrentBountyComments } from "contexts/reducers/change-current-bounty";
+import { addToast } from "contexts/reducers/change-toaster";
 
-import {MetamaskErrors} from "interfaces/enums/Errors";
-import { NetworkEvents } from "interfaces/enums/events";
-import {pullRequest} from "interfaces/issue-data";
+import { issueParser } from "helpers/issue";
 
+import { MetamaskErrors } from "interfaces/enums/Errors";
+import { NetworkEvents } from "interfaces/enums/events";
+import {
+  IssueBigNumberData,
+  IssueData,
+  pullRequest,
+} from "interfaces/issue-data";
+
+import {
+  getBountyData,
+  getBountyOrPullRequestComments,
+  getPullRequestReviews,
+  getPullRequestsDetails,
+} from "x-hooks/api/get-bounty-data";
 import useApi from "x-hooks/use-api";
 import useBepro from "x-hooks/use-bepro";
-import {useBounty} from "x-hooks/use-bounty";
-import {useNetwork} from "x-hooks/use-network";
+import { useNetwork } from "x-hooks/use-network";
 
-export default function PullRequestPage() {
-  const router = useRouter();
+interface PagePullRequestProps {
+  bounty: IssueData;
+  pullRequest: pullRequest;
+  _nextI18Next?: SSRConfig;
+}
 
+export default function PullRequestPage({ pullRequest, bounty }: PagePullRequestProps) {
   const { t } = useTranslation(["common", "pull-request"]);
 
   const [showModal, setShowModal] = useState(false);
   const [isCancelling, setIsCancelling] = useState(false);
   const [isMakingReady, setIsMakingReady] = useState(false);
-  const [pullRequest, setPullRequest] = useState<pullRequest>();
+  const [currentBounty, setCurrentBounty] = useState<IssueBigNumberData>(issueParser(bounty));
```

Comment:
why do we keep doing this parsing? wouldn't make this data parsed from the start be more easy than having to remember we need to parse it whenever we want to read it? :o